utils/notifications.py

```
@@ -0,0 +1,211 @@
+                "color": color,
+                "text": {"title": title, "content": body},
+                "ids": {"channel": self.notifiarr_channel},
+            },
+        }
+
+        headers = {"Content-Type": "application/json", "X-API-Key": self.notifiarr_key}
+
+        def send():
+            """Inner function for retry logic."""
+            data = json.dumps(payload).encode("utf-8")
+            req = urllib.request.Request(url, data=data, headers=headers, method="POST")
```

Ruff (0.14.6)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)